### PR TITLE
Functionality for flattening objects

### PR DESCRIPTION
**Pull request by konstantpapp**
Flatten JSON objects

### CODE REVIEW

**Comment by yamafaktory** [comment]
Hey @konstantpapp !

Thanks for submitting a pull-request.

I have a few concerns about what you're introducing here since flattening an object might not be as straightforward as it seems to be at first glance:

- Why renaming the target key as a *path*? If e.g. we have `{ "foo": { "bar": { "super-super-super-super-super-long-key": { "even-longer-super-super-super-super-super-long-key": "something" } } } }`, we will end up with `{ "foo.bar.super-super-super-super-super-long-key.even-longer-super-super-super-super-super-long-key": "something" }`. This can get even more cumbersome if an object has a lot of nested children.
- I might be wrong, but this is not going to work with nested objects with more than one property: `{ "foo": { "bar": 1, "woot": 2 } }`

Is this feature coming from `jq` (I tried not to mimic it)? I have checked but I haven't found any flag or special grammar fro that (I'd assume this is related to the fact that this can't be done properly, as for arrays). 

I'm pretty open to additional if they don't bloat the scope of this simple CLI. However in this case, I'm a bit confused to be honest 😄.

**Comment by konstantpapp** [comment]
Hi @yamafaktory and many thanks for the feedback! 😀

I got this idea for this feature as I felt that it could help read key-value pairs in objects with many nested properties. It works with nested objects with more than one property:
```
{ "foo": { "bar": 1, "woot": 2 } }
````
wil give
```
{
  "foo.bar": 1,
  "foo.woot": 2
}
```
I don't know if `jq` supports this feature. I agree though that it looks extra verbose in objects with long keys.

**Comment by yamafaktory** [comment]
Hey @konstantpapp, thanks for the feedback!

I get another question regarding "mixed" levels of nesting, let's say e.g.:

```json
{ "foo": { "bar": 1, "woot": { "a": 2, "b": 3 } } }
```

What will happen?

File: example.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (yamafaktory) [inline example.json:1]: We can drop this file.

**Comment by konstantpapp** [comment]
> Hey @konstantpapp, thanks for the feedback!
> 
> I get another question regarding "mixed" levels of nesting, let's say e.g.:
> 
> ```json
> { "foo": { "bar": 1, "woot": { "a": 2, "b": 3 } } }
> ```
> 
> What will happen?

We would get the following:
```
{
  "foo.bar": 1,
  "foo.woot.a": 2,
  "foo.woot.b": 3
}
```

**Comment by yamafaktory** [comment]
Thanks a ton @konstantpapp 🍻  for your contribution and this new feature!

I'll create a new release later today 🚀 .

**Pull request merged**